src/dhis2.angular.services.js

```diff
@@ -252,6 +254,51 @@ var d2Services = angular.module('d2Services', ['ngResource'])
             dateValue = moment(dateValue, 'YYYY-MM-DD')._d;
             return $filter('date')(dateValue, calendarSetting.keyDateFormat);
         },
+        formatFromApiToUserCalendar: function (dateValue) {
+            if (!dateValue) {
+                return;
+            }
+
+            var calendarSetting = CalendarService.getSetting();
+
+            //A bit hacky way to check if format id dd-mm-yyyy.
+            if(dateValue.charAt(2) === '-') {
+                dateValue = moment(dateValue, calendarSetting.momentFormat)._d;
+                dateValue = Date.parse(dateValue);
+                dateValue = $filter('date')(dateValue, 'yyyy-MM-dd');
+            }
+
+            var splitDate = dateValue.split('-');
+
+            //Months are for some reason 0 based.
+            var date = new Date(splitDate[0], splitDate[1]-1, splitDate[2]);
+
+            if(calendarSetting.keyCalendar === 'ethiopian') {
+                date = date.toLocaleDateString('en-GB-u-ca-ethiopic');
+
+            } else if(calendarSetting.keyCalendar === 'coptic') {
+                date = date.toLocaleDateString('en-GB-u-ca-coptic');
+
+            } else if(calendarSetting.keyCalendar === 'gregorian') {
+                date = date.toLocaleDateString('en-GB-u-ca-gregory');
+
+            } else if(calendarSetting.keyCalendar === 'islamic') {
+                date = date.toLocaleDateString('en-GB-u-ca-islamic');
+                
+            } else if(calendarSetting.keyCalendar === 'iso8601') {
+                date = date.toLocaleDateString('en-GB-u-ca-iso8601');
+            
+            } else if(calendarSetting.keyCalendar === 'persian') {
+                date = date.toLocaleDateString('en-GB-u-ca-persian');
+                
```

Comment:
@lukasewvd It looks like you are missing two calendars here (Nepali and Thai). This is not however directly related to this particular bug I think, but just wanted to point it out.

Reply:
Good point, I will add them. Thanks for pointing it out.

Reply:
Nepali does not seem to be an option from [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleDateString), but I think "Buddhist" calendar here refers to the same which we rather confusingly call "Thai" http://joda-time.sourceforge.net/cal_buddhist.html